debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java

```diff
@@ -107,15 +87,72 @@
             // Default is prefixed with "__dbz__" to minimize the likelihood of a conflict with an existing key field name.
             .withDefault("__dbz__physicalTableIdentifier")
             .withDescription("Each record's key schema will be augmented with this field name. The purpose of this " +
-                    "field is to distinguish the different physical tables that can now share a single topic. Make " +
-                    "sure not to configure a field name that is at risk of conflict with existing key schema field " +
-                    "names.");
+                "field is to distinguish the different physical tables that can now share a single topic. Make " +
+                "sure not to configure a field name that is at risk of conflict with existing key schema field " +
+                "names.");
+    private static final Field KEY_FIELD_REPLACEMENT = Field.create("key.field.replacement")
+            .withDisplayName("Key field replacement")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.LONG)
+            .withImportance(ConfigDef.Importance.LOW)
+            .withValidation(ByLogicalTableRouter::validateKeyFieldReplacement)
+            .withDescription("The replacement string used in conjunction with " + KEY_FIELD_REGEX.name() +
+                ". This will be used to create the physical table identifier in the record's key.");
+
+    private static final Logger logger = LoggerFactory.getLogger(ByLogicalTableRouter.class);
+    private final AvroValidator schemaNameValidator = AvroValidator.create(logger);
+    private Pattern topicRegex;
+    private String topicReplacement;
+    private Pattern keyFieldRegex;
+    private String keyFieldReplacement;
+    private String keyFieldName;
+    private final Cache<Schema, Schema> keySchemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
+    private final Cache<Schema, Schema> envelopeSchemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
+
+    /**
+     * Logic as follows:
+     * If KEY_FIELD_REGEX has a value that is really a regex, then the KEY_FIELD_REPLACEMENT must be a non-null value.
+     */
+    private static int validateKeyFieldReplacement(Configuration config, Field field, Field.ValidationOutput problems) {
+        String keyFieldRegex = config.getString(KEY_FIELD_REGEX);
+        if (keyFieldRegex != null) {
+            keyFieldRegex = keyFieldRegex.trim();
+        }
+
+        String keyFieldReplacement = config.getString(KEY_FIELD_REPLACEMENT);
+        if (keyFieldReplacement != null) {
+            keyFieldReplacement = keyFieldReplacement.trim();
+        }
+
+        if (keyFieldRegex == null && keyFieldReplacement == null) {
+            return 0;
+        }
+
+        if (!Objects.equals(keyFieldRegex, "")
```

Comment:
That would mean that `keyFieldRegex` is guaranteed to be not null, which it is not.